File: Assignment(G2)/A4_108502516/A4_108502516.java
/*
 * Assignment 4
 * Course: CE1004-A
 * Name: ¡¬À»È∫
 * Student ID: 108502516
 */
package hello;
import java.util.Scanner;
import java.util.Arrays;

public class A4_108502516 {

	public static void main(String[] args) {

		//Scanner
		Scanner input = new Scanner(System.in);
		
		//input the size of map
		String Size = input.nextLine();
		int size = Integer.parseInt(Size);
		
		//declare map array
		String[][] arr = new String[size+2][size+2];
		
		//replace null elements into 0
		for(int i=0;i<=size+1;i++)
			Arrays.fill(arr[i], "0");
		
		//input mine map into array
		for(int i=1;i<=size;i++)
		{
			String line = input.nextLine();
			for(int x=1;x<=line.length();x++)
			{
				arr[i][x] = Character.toString( line.charAt(x-1) );
			}
		}
		
		//judge the elements in array are mine or not
		for(int i=1;i<=size;i++)
		{
			for(int x=1;x<=size;x++)
			{
				//if the element is mine, skip
				if(arr[i][x].equals("*"))
				{
					continue;
				}
				
				//replace "_" into the number of mines around
				else if(arr[i][x].equals("_"))
				{
					int num = 0;
					for(int a=i-1;a<=i+1;a++)
					{
						for(int b=x-1;b<=x+1;b++)
						{
							if(arr[a][b].equals("*"))
								num++;
						}
					}
					arr[i][x] = Integer.toString(num);
				}
				
			}
		}
		
		//print array
		for(int i=1;i<=size;i++)
		{
			for(int x=1;x<=size;x++)
			{
				if(x==size)
					System.out.println(arr[i][x]);
				else
					System.out.print(arr[i][x]);
			}
		}
		
		//close input
		input.close();
	}

}

File: Assignment/A4/A4-108502516.cpp
/*
*   Assignment 4
*   Name: ÁÂËÈéº
*   Student Number: 108502516
*   Course: 2019-CE1003-A
*/
#include<iostream>
#include<iomanip>

using namespace std;

void find_f(int);
void find_p(int);

int main()
{
    int num;
    cin >> num;
    while(num<0)
        cin >> num;

    find_f(num);
}

void find_f(int num)
{
    int a, b;
    for(a=1;a<=num;a++)
    {
        b=num%a;
        if(b==0)
        {
            cout << a << " ";
            find_p(a);
        }
    }
    return;
}

void find_p(int num)
{
    int a, b, Count=0;
    for(a=1;a<=num;a++)
    {
        b=num%a;
        if(b==0)
            Count++;
    }
    if(Count>2||Count==1)
        cout << "n" << endl;
    else if(Count==2)
        cout << "y" << endl;
}
